### ISSUE TITLE
Fix lock/unlock error in Google Assistant with HA cloud

### ISSUE THREAD

**Issue by Kevr0n228** [fuatakgun/kia_uvo#148]
I just signed up for the Home Assistant Cloud trial and linked it with Google Home, and GH automatically recognized the lock/unlock action and allowed me to use it through a voice command. The command works and the car locks/unlocks, but google always says "something went wrong" after doing it. 

Super low priority, but just wanted to put it out there.

**Comment by dahlb** [comment]
@Kevr0n228 google actions require a response time of under 5 seconds; the car doesn't finish that fast so google says something went wrong because it stops waiting at 5 seconds, but the car will continue processing the request which is why is succeeds

**Comment by Kevr0n228** [comment]
Thanks for the info! I didn't know that. Since it'll be a cold day in hell when bluelink takes less than 5 seconds to do anything besides crash, I'm gonna close this.

**Comment by cdnninja** [comment]
Quick ways around this for your @Kevr0n228.   Create a "help" input boolean named what you want to tell google.  Create an automation that when this switch turns on it calls the service you would like and turns the switch back off.    This will also work on remote start / start climate.    You have to expose input_booleans to google but works great once that is done.   Google can't tell you if something went wrong though as a note, not that it can today.

**Comment by fuatakgun** [comment]
Very helpful discussion, as an improvement point, you can set another automation to announce to google speaker or send a notification to your mobile phone when car is actually locked or could not lock in 5 minutes, so you are eventually aware.

**Comment by dahlb** [comment]
@Kevr0n228 

another solution I use for climate is to make a synthetic switch for google home to seamlessly interact with, I don't think I've seen my input boolean's in GA, though I use the community version instead of HA's official one.  it's important to use script.turn_on instead of invoking the script directly so it doesn't wait for completion.

configuration.yaml

```
switch: !include_dir_list switch
```

switch/car_climate.yaml 
```
platform: template
switches:
  car_climate:
    friendly_name: Car Climate
    value_template: "{{ is_state('binary_sensor.niro_ev_air_conditioner','on') }}"
    turn_on:
      service: script.turn_on
      target:
        entity_id: script.car_climate
    turn_off:
      service: kia_uvo.stop_climate
```

turn on script
```
car_climate:
  alias: Car Climate
  sequence:
  - choose:
    - conditions:
      - condition: device
        device_id: 10dfc81975715f86cc3eb4f48fb3a519
        domain: lock
        entity_id: lock.niro_ev_door_lock
        type: is_unlocked
      sequence:
      - service: lock.lock
        target:
          entity_id: lock.niro_ev_door_lock
    default: []
  - service: kia_uvo.start_climate
    data:
      Duration: 10
      Temperature: 76
      Heating: true
  mode: single
```

One of my main objectives with this integration was to fix the app doesn't auto lock the doors in order to turn on the climate, it just leaves you to lock them manually so the climate takes a long time to get on through the app having to wait for each of those commands to finish and picking the temperature settings every time.

**Comment by Kevr0n228** [comment]
This is likely the way I'll go about it. I already have scripts to start
and stop climate and was using a GH routine to call it a la: "hey google,
heat up the car" for which google will run the command "turn on start kona
climate" which triggers my script `start_kona_climate`. Yours is a more
robust solution and includes the locking (which I completely forgot
about!). Thanks for the ideas!
